realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -1947,6 +1947,36 @@ public void onChange() {
         result.load();
     }
 
+    @Test
+    @RunTestInLooperThread
+    public void testAsyncTransactionWorksWithAsyncQuery() {
+        RealmResults<AllTypes> results = looperThread.realm.where(AllTypes.class).findAllAsync();
```

Comment:
A pattern we have used elsewhere is:
```
Realm realm = looperThread.realm;
```

It makes it slightly more readable, and it is easier to replace the Realm instance if we need to.